Add interaction indicators to List View and block UI

kyrapieterse opened the pull request:
Introduces a blue dot indicator in the List View for blocks with interactions by adding a data attribute via a new listViewObserver extension. Updates styles to highlight blocks with interactions in both the editor and List View, and adjusts MaxiBlock logic to apply the interaction class when relations exist. Minor style tweaks and refactoring are included to support these changes.

# Description

<!---
Please include a summary of the changes and the related issue.
Please also include relevant motivation and context.
--->

Fixes # (issue)

# How Has This Been Tested?

<!---
Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.
--->

# Test checklist

<!--- Please remove the unnecessary checkbox --->

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Editor shows block boundary accents, list-view dots and breadcrumb markers for interaction and non-color background layers; block state now drives these visuals.

* **Style**
  * Active/selected interaction uses pink and bold text; background uses yellow. Relation buttons, breadcrumbs spacing, hover and active states refined.

* **UI Tokens**
  * Added interaction/background color tokens (and related tokens); removed a couple legacy color tokens.

* **Bug Fixes**
  * List-view observation and cleanup improved for more consistent, performant indicator updates.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kyrapieterse committed:
Add interaction indicators to List View and block UI

Introduces a blue dot indicator in the List View for blocks with interactions by adding a data attribute via a new listViewObserver extension. Updates styles to highlight blocks with interactions in both the editor and List View, and adjusts MaxiBlock logic to apply the interaction class when relations exist. Minor style tweaks and refactoring are included to support these changes.

kyrapieterse committed:
Add interaction indicator styles and attributes

Introduces a new --maxi-interaction-pink variable and updates all relevant styles to use it for interaction indicators. Breadcrumbs and list view now visually highlight blocks with interactions, and aria-label/title attributes are added for accessibility. This improves clarity and accessibility for users identifying interactive blocks.

kyrapieterse committed:
Remove debug logs and improve ListView observer cleanup

Removed unnecessary debug logging from breadcrumbs and ListView observer. Refactored ListView observer to track and clean up resources (timeouts, intervals, subscriptions) and exposed a cleanup function for manual teardown. Updated interaction indicator color in backend variables.

kyrapieterse committed:
Add background layer indicators to UI components

Introduces yellow indicators for blocks with non-color background layers across the editor, breadcrumbs, and list view. Updates SCSS and JS logic to support visual cues and accessibility labels for background layers, alongside existing interaction indicators.

elzadj committed:
merge master and fix growing bg labels

elzadj committed:
Fix styles for 2 dots when bg and ib are together on the same block

Olekrut reviewed (changes requested):
1) Tooltips look weird
<img width="439" height="230" alt="image" src="https://github.com/user-attachments/assets/f8d3bfc7-3a9a-4cb6-88df-474304871e70" />
2) UX this dot - not sure that will help a lot for users who dont know it
Maybe need to make it more noticeble, like small 2 letter (IB) inside circle.
And to add description to outline tab if possible
<img width="421" height="485" alt="image" src="https://github.com/user-attachments/assets/8ebf119b-68f5-4cd0-867f-5ebe6ad7533f" />

elzadj commented:
We can just add a legend that explains what the dots mean. At the bottom of the list view, it has a lot of space

kyrapieterse committed:
Remove redundant title attributes from list view items

Eliminated setting the 'title' attribute on list view items, relying on 'aria-label' for accessibility. This reduces redundancy and streamlines attribute management.
